src_erl/Communication_Layer/http_Nerlserver/src/Client/clientStatem.erl

```diff
@@ -311,71 +310,79 @@ code_change(_OldVsn, StateName, State = #client_statem_state{}, _Extra) ->
-  LayersSizes = nerl_tools:string_to_list_int(maps:get(<<"layersSizes">>,Worker)),
-  LayersActivationFunctions = nerl_tools:string_to_list_int(maps:get(<<"layersActivationFunctions">>,Worker)),
-
-  FederatedMode = list_to_integer(binary_to_list(maps:get(<<"federatedMode">>,Worker))),
-  CountLimit = list_to_integer(binary_to_list(maps:get(<<"countLimit">>,Worker))),
-  Optimizer = list_to_integer(binary_to_list(maps:get(<<"optimizer">>,Worker))),
-  Features = list_to_integer(binary_to_list(maps:get(<<"features">>,Worker))),
-  Labels = list_to_integer(binary_to_list(maps:get(<<"labels">>,Worker))),
-  LossMethod = list_to_integer(binary_to_list(maps:get(<<"lossMethod">>,Worker))),
-  LearningRate = list_to_float(binary_to_list(maps:get(<<"learningRate">>,Worker))),
-  %LearningRate = list_to_float(binary_to_list(maps:get(<<"learningRate">>,Worker))),
-
-  WorkerArgs = {WorkerName,ModelId,ModelType,ScalingMethod
-                , LayerTypesList, LayersSizes
-                , LayersActivationFunctions, FederatedMode
-                , CountLimit, Optimizer, Features, Labels, LossMethod, LearningRate,  self() },
-  WorkerPid = nerlNetStatem:start_link(WorkerArgs),
-  % timingMap = #{{WorkerName1=>{LastBatchReceivedTime,totalBatches,AverageTrainingime},{Worker2,..}, ...}
-  createWorkers(Workers,WorkerModelID+1,ClientPid,maps:put(WorkerName, WorkerPid,WorkersNamesPidsMap),maps:put(WorkerName,{0,0,0.0},TimingMap)).
+createWorkers(ClientName, EtsRef) ->
+  CLIENT_WORKES_MAPS_TUPLE_IDX = 3,
+  MapOfHostClients = maps:from_list(ets:lookup_element(nerlnet_data, hostClients, ?DATA_IDX)), % This is the format of hostClients {Name,{Port,ClientWorkers,ClientWorkersMaps}}
+  ClientWorkersMaps = element(CLIENT_WORKES_MAPS_TUPLE_IDX,maps:get(ClientName, MapOfHostClients)),
+  Func = fun(WorkerMap) -> 
+    ModelId = erlang:unique_integer([positive]),
+    %%  TODO: move to json parser
+    WorkerName = list_to_atom(binary_to_list(maps:get(<<"name">>,WorkerMap))),
+    ModelType = list_to_integer(binary_to_list(maps:get(<<"modelType">>,WorkerMap))),
+    ScalingMethod = list_to_integer(binary_to_list(maps:get(<<"scalingMethod">>,WorkerMap))),
+
+    LayerTypesList = nerl_tools:string_to_list_int(maps:get(<<"layerTypesList">>,WorkerMap)),
+    LayersSizes = nerl_tools:string_to_list_int(maps:get(<<"layersSizes">>,WorkerMap)),
+    LayersActivationFunctions = nerl_tools:string_to_list_int(maps:get(<<"layersActivationFunctions">>,WorkerMap)),
+
+    Optimizer = list_to_integer(binary_to_list(maps:get(<<"optimizer">>,WorkerMap))),
+    Features = list_to_integer(binary_to_list(maps:get(<<"features">>,WorkerMap))),
+    Labels = list_to_integer(binary_to_list(maps:get(<<"labels">>,WorkerMap))),
+    LossMethod = list_to_integer(binary_to_list(maps:get(<<"lossMethod">>,WorkerMap))),
+    LearningRate = list_to_float(binary_to_list(maps:get(<<"learningRate">>,WorkerMap))),
+
+    % TODO add documentation about this case of 
+    case ModelType of
```

Comment:
@halfway258 
I think this case ModelTypeOf should have a dedicated function with documentation even a module called client controller. if someone wants to add a new worker then this is the part he should change and it will be easier in a separate file.